libs/knowledge-store/tests/integration_tests/test_graph_store.py

```diff
@@ -195,6 +195,17 @@ def test_mmr_traversal(
     results = gs.mmr_traversal_search("0.0", k=4, metadata_filter={"even": True})
     assert _result_ids(results) == ["v0", "v2"]
 
+    # with neighborhood=[v0], we should start traversal there. this means that
+    # the initial candidates are `v2`,`v3`. `v1` is unreachable and not
+    # included.
+    results = gs.mmr_traversal_search("0.0", k=4, neighborhood=["v0"])
+    assert _result_ids(results) == ["v2", "v3"]
```

Comment:
What would the code look like to re-use results? Would you need to map the ID out of the returned results set? Would it make sense to accept documents as the result set, and then pull the IDs out inside the implementation, or maybe provide a helper method to translate a set of documents into a set of IDs?

Reply:
I don't think we should try to over-beautify this yet. We're providing components for people to write code, and they should be able to write the for-comprehension to do `[d.id for d in results]`. I also don't think we want to take documents, because they may want to store just the IDs in state somewhere, etc. We don't want to dictate they nede the whole document.